Earthfile

```diff
@@ -39,250 +59,368 @@ setup:
+  COPY scripts/generate-repos.sh scripts/merge-repos.py ${SPACEROS_DIR}/scripts/
+  RUN chmod +x ${SPACEROS_DIR}/scripts/generate-repos.sh ${SPACEROS_DIR}/scripts/merge-repos.py \
+    && mv ${SPACEROS_DIR}/rosdeps.sh ${SPACEROS_DIR}/scripts/rosdeps.sh
+
+  # Post Installation
+  DO +POST_INSTALLATION --IMAGE_VARIANT=${IMAGE_VARIANT}
+
+  # Clear Apt and Pip cache
+  RUN rm -rf /var/lib/apt/lists/* /tmp/pip-reqs /var/cache/apt/archives \
+    && apt-get clean \
+    && pip cache purge
+
+  # Add user and group
+  RUN useradd -m -s /bin/bash ${USERNAME} \
+    && chown -R ${USERNAME}:${USERNAME} ${SPACEROS_DIR}/scripts \
+    && echo "${USERNAME} ALL=(ALL) NOPASSWD:ALL" > /etc/sudoers.d/${USERNAME}
+  RUN echo "source ${SPACEROS_DIR}/setup.bash" >> ${HOME}/.bashrc
+
+  RUN chown -R ${USERNAME}:${USERNAME} ${HOME}
+
+  USER ${USERNAME}
+  WORKDIR ${HOME}
+
+  # Only save image if build-test is successful
+  IF [ "${SKIP_BUILD_TEST}" = "true" ]
+    SAVE IMAGE --push ${IMAGE_NAME}:${IMAGE_VARIANT}
+  ELSE
+    BUILD +build-test
+    SAVE IMAGE --push ${IMAGE_NAME}:${IMAGE_VARIANT}
+  END
+
+###############################################################################
+### Post Installation Stage
+# This stage is responsible for cleaning up the workspace and installing
+# additional dependencies based on the image variant.
+###############################################################################
+POST_INSTALLATION:
+  FUNCTION
+  ARG --required IMAGE_VARIANT
+
+  # Remove workspace if image variant is core or dev
+  RUN rm -rf ${WORKSPACE_DIR}
+
+  # If Dev, install IKOS and excluded dependencies
+  IF [ "${IMAGE_VARIANT}" = "dev" ]
+    DO +ADD_IKOS
+
+    COPY excluded-deps.txt ./
+    RUN apt install -y $(grep -v '^#' excluded-deps.txt) \
+      && rm -rf excluded-deps.txt
+  END
+
+###############################################################################
+### Push Image Stage for core image
+# This stage is responsible for pushing the core image to the registry.
+###############################################################################
+push-core-image:
+  # This can be overridden with a blank string to prevent pushing to the registry.
+  ARG --required VCS_REF
+  ARG TAG
+
+  FROM +image --IMAGE_VARIANT=core
 
-  WORKDIR ${HOME_DIR}
-  COPY docker/entrypoint.sh /ros_entrypoint.sh
-  ENTRYPOINT ["/ros_entrypoint.sh"]
-  CMD ["bash"]
-  SAVE IMAGE osrf/space-ros:${VERSION}
+  LABEL org.label-schema.schema-version="1.0"
+  LABEL org.label-schema.name="Space ROS - Core"
+  LABEL org.label-schema.description="Core version of the Space ROS platform"
+  LABEL org.label-schema.vendor="Open Robotics"
+  LABEL org.label-schema.url="https://github.com/space-ros"
+  LABEL org.label-schema.vcs-url="https://github.com/space-ros/space-ros"
+  LABEL org.label-schema.vcs-ref=${VCS_REF}
 
-# Target for prepping image(s) to be pushed to remote registries.
-push-image:
-  FROM +image
+  SAVE IMAGE --push ${IMAGE_NAME}:core ${TAG}
```

Comment:
Forgive my ignorance here, but with my understanding of naming will this resolve to something like:

```
SAVE IMAGE --push space-ros:core osrf/space-ros:main
```

Is that valid syntax for pushing to remote registries? I think I ran into something that prevent that, hence breaking out the `SAVE IMAGE --push` to its own separate step with a single tag.

Reply:
I think you have spotted a problem. I will fix it.